core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ConsumerBindingService.java

```diff
@@ -23,94 +25,81 @@
+      throw new ValidationException("Can't update " + consumerBinding.getKey() + " because it doesn't exist");
     }
+    consumerBindingValidator.validateForUpdate(consumerBinding, existing.get());
+    consumerBinding.setSpecification(handlerService.handleUpdate(consumerBinding, existing.get()));
+    return save(consumerBinding, UPDATE);
+  }
 
-    consumerBindingValidator.validateForUpdate(consumerBinding, dataToModel.convertToModel(existing.get()));
-    consumerBindingData.setSpecification(modelToData.convertToData(handlerService.handleUpdate(consumerBinding, dataToModel.convertToModel(existing.get()))));
-    ConsumerBinding out = dataToModel.convertToModel(consumerBindingRepository.save(consumerBindingData));
-    consumerBindingServiceEventEmitter.emitEventOnProcessedEntity(EventType.UPDATE, out);
-    return Optional.ofNullable(out);
+  private Optional<ConsumerBinding> save(ConsumerBinding consumerBinding, EventType eventType) {
+    consumerBinding = consumerBindingRepository.save(consumerBinding);
+    consumerBindingServiceEventEmitter.emitEventOnProcessedEntity(eventType, consumerBinding);
+    return Optional.ofNullable(consumerBinding);
   }
 
   public Optional<ConsumerBinding> upsert(ConsumerBinding consumerBinding) throws ValidationException {
-    return !consumerBindingRepository.findById(modelToData.convertToData(consumerBinding).getKey()).isPresent() ?
-        create(consumerBinding) :
-        update(consumerBinding);
+    return !read(consumerBinding.getKey()).isPresent() ? create(consumerBinding) : update(consumerBinding);
   }
 
-  public void delete(ConsumerBinding consumerBinding) {
-    throw new UnsupportedOperationException();
+  public Optional<ConsumerBinding> read(ConsumerBindingKey key) {
+    return consumerBindingRepository.findById(key);
   }
 
   public List<ConsumerBinding> findAll(Predicate<ConsumerBinding> filter) {
-    return consumerBindingRepository.findAll().stream().map(dataToModel::convertToModel).filter(filter).collect(toList());
+    return consumerBindingRepository.findAll().stream().filter(filter).collect(toList());
   }
 
-  public Optional<ConsumerBinding> find(ConsumerKey key) {
-    var example = new
-        ConsumerBindingData(
-        new ConsumerBindingDataKey(
-            key.getStreamDomain(),
-            key.getStreamName(),
-            key.getStreamVersion(),
-            key.getZone(),
-            null,
-            key.getName()
-        ), null, null);
-    return consumerBindingRepository.findAll(Example.of(example)).stream().findFirst().map(dataToModel::convertToModel);
+  public void delete(ConsumerBinding consumerBinding) {
+    throw new UnsupportedOperationException();
   }
 
   public boolean exists(ConsumerBindingKey key) {
     return read(key).isPresent();
   }
+
+  public Optional<ConsumerBinding> find(ConsumerKey key) {
+    var example = new ConsumerBinding(new ConsumerBindingKey(
```

Comment:
We should consider exposing find(example) or find(someLookupModel) here so that we have a standard way to do all finds. e.g. by passing a GraphQL free object to 'filter by'